Physical_Engine/Query/Geometry.cs

```diff
@@ -59,6 +62,41 @@ public static oM.Geometry.ISurface Geometry(this oM.Physical.Elements.ISurface s
 
         /***************************************************/
 
+        [Description("Gets the geometry of a PadFoundation as a surface. Method required for automatic display in UI packages.")]
+        [Input("padFoundation", "PadFoundation to get the surface geometry from.")]
+        [Output("surface", "The surface defining the PadFoundation.")]
+        public static IGeometry Geometry(this PadFoundation padFoundation)
+        {
+            return padFoundation.IsNull() ? null : padFoundation.Location;
+        }
+
+        [Description("Gets the geometry of a PileFoundation as a surface representing the pile cap and curves representing the piles. Method required for automatic display in UI packages.")]
+        [Input("pileFoundation", "PileFoundation to get the geometry from.")]
+        [Output("geometry", "The geometry defining the PadFoundation.")]
+        public static IGeometry Geometry(this PileFoundation pileFoundation)
+        {
+            List<IGeometry> geometry = new List<IGeometry>();
+            IGeometry pileCapGeometry = pileFoundation.PileCap.Geometry();
+            if(!pileCapGeometry.IsNull())
```

Comment:
I would rather simple check `== null` here - otherwise you will get an error plus a warning in case of a null geometry